web/src/data/api.ts

```diff
@@ -685,11 +685,12 @@ async function me(): Promise<User | null> {
   }
 }
 
-async function getAllUsers(params: SearchParams) {
+async function getAllUsers(params: SearchParams, input_search_filter: String | null) {
```

Comment:
```suggestion
async function getAllUsers(params: SearchParams, inputSearchFilter: string | null) {
```

If you _wanted_ to, you could probably add something to the `SearchParams` interface to capture a filter string. It looks like that interface is currently just a bunch of optional stuff anyway, and can be used with various types of "search" endpoints. Not necessary, but might be useful.

If you want to keep `input_search_filter`, I think it would be best to switch to camelCase (i.e. `inputSearchFilter`) for the parameter, since we're in typescript land. I also think the type should be `string` instead of `String`. The [typescript docs](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#number-string-boolean-symbol-and-object) recommend against using the object wrapper classes for primitive types.

Reply:
I started working on this, and it was causing some strange issues on front end things like the number of rows per page being maxed out no matter what and the next page skipping around strangely. 

For now I think I'll leave this portion as is, but it might be worth making that change in the future if we have more things we want a filter string for.

I did change the casing for the variable name and changed `String` to `string`